### DISCUSSION TITLE
`Accordion.Trigger` getting `React.Children.only` error wen used with `asChild`

### DISCUSSION THREAD

**Discussion by AndreaBarghigiani** [radix-ui/primitives#2594] [Help] [answered]
I am looking to integrate the `Accordion` component in my app, but since I do not want to have an `h3` as a wrapper I thought to leverage the `asChild` prop to let it use a more general `div` element, but every time I try to use it I get the `React.Children.only` error. 

I tried to check the source code, but it has too many inheritance and make it too complex for me in order to understand how to propose a solution 😊

**Comment by jjenzz** [comment, marked as answer]
the `h3` is rendered by the `Accordion.Header` part https://codesandbox.io/p/devbox/youthful-tess-xq95js?file=%2FApp.jsx%3A33%2C1-34%2C1.

if you are putting `asChild` on `Accordion.Trigger`, you're error will likely be occurring because it is trying to attach itself to the text within the `button`, which is not valid.
